src/main/jbake/content/blog/2016/07/14/hawkular-apm-openshift.adoc

```diff
@@ -0,0 +1,123 @@
+:jbake-tags: blog, btm, apm, distributed tracing, application performance, business transactions, openshift, cloud, docker
+
+This post will show how to use the Hawkular APM project, previously known as Hawkular BTM, to monitor an application running within Openshift. To achieve this both the Hawkular APM server, and the application (a vert.x Javascript example), will be deployed as Docker images.
+
+We will use the Openshift CLI, and therefore will need to login. For example, to login into the Openshift CDK environment:
+
+----
+oc login 10.1.2.2:8443 -u openshift-dev -p devel
+----
+
+=== Deploy Hawkular APM Server
+
+The first step is to deploy the Hawkular APM server. There are two ways that this can be achieved.
+
+==== Image from Docker Hub
+
+The Hawkular APM server has a docker image registered at https://hub.docker.com/r/jboss/hawkular-apm-server[Docker Hub]. This can be deployed to Openshift using the following commands:
+
+----
+oc new-app jboss/hawkular-apm-server
+oc expose service hawkular-apm-server
+----
+
+The _new-app_ command deploys the image as a service, and the _expose_ command makes it available via a stable URI.
+
+NOTE: If the deploy fails with an "unknown manifest" error in the openshift events log, this is due to a http://lists.openshift.redhat.com/openshift-archives/users/2016-June/msg00054.html[known docker issue related to a change between version 1.9 and 1.10]. In this case, use the local deploy approach described below.
+
+==== Local Image
+
+To deploy the APM server from a local build, it will be necessary to get the https://github.com/jboss-dockerfiles/hawkular/blob/8ffb752b68c0f49a2150fe345e660c53c32fb954/hawkular-apm-server/Dockerfile[_Dockerfile_]. Place this file in a suitable folder and from that location run the following:
+
+----
+oc new-build --binary --name=hawkular-apm-server
+oc start-build hawkular-apm-server --from-dir=. --follow
+oc new-app hawkular-apm-server
+oc expose service hawkular-apm-server
+----
+
+==== In the Openshift Console
+
+Once either of these options has been used to deploy the Hawkular APM Server, it can be seen within the Openshift console.
+
+.Hawkular APM Server deployed in Openshift
+ifndef::env-github[]
+image::/img/blog/2016/2016-07-14-osapmserver.png[Hawkular APM Server]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/2016-07-14-osapmserver.png[Hawkular APM Server]
+endif::[]
+
+
+
+=== Deploy Example Vert.x Javascript Application
+
+The application that we will monitor is a Vert.x Javascript example application. Either clone the https://github.com/vert-x3/vertx-examples[github repository] or simply copy the two relevant files from the https://github.com/vert-x3/vertx-examples/tree/master/docker-examples/vertx-docker-javascript[vertx-examples/docker-examples/vertx-docker-javascript] folder.
+
+To enable the application to be instrumented by Hawkular APM, it will be necessary to add the following lines, before the _EXPOSE 8080_ line:
```

Comment:
it's rather obvious but you could mention that you are talking about: Dockerfile file.